### ISSUE TITLE
how to test an yet to be published version of build package?

### ISSUE THREAD

**Issue by maks** [dart-lang/build#2579]
I'd like to be able to test out the new support for `assets` added in #2573  but if I try to specify a git url for `build_runner` package in a flutter app's pubspec.yaml along with the package (string_res) where I have my builder as either a git or path in the apps pubspec I get an error that:
```
because xyz_flutter_app depends on both build_runner from git and string_res from git, version solving failed.
```
Is there a way to do this that I'm missing?

**Comment by natebosch** [comment]
See https://dart.dev/tools/pub/dependencies#dependency-overrides

If you specify a `dependency_overrides:` section with this git dependency it should stop complaining about other non-git dependencies on the same pacakge.

**Comment by jakemac53** [comment]
Note that this has been published in `build_runner_core` version `4.3.0`
